content/api/commands/session.md

```diff
@@ -340,13 +265,27 @@ it('should transfer money between users', () => {
 
 ### Validating the session
 
-If the `validate` function return `false`, throws an exception, returns a
-Promise that resolves to `false` or rejects, or contains any failing Cypress
-command, the session will be considered invalid, and `setup` will be re-run.
+The `validate` function has a slightly different function depending on whether
+it is being run immediately after `setup`, or after restoring the saved session.
+
+After `setup`, the session is not saved until `validate` has returned or
+resolved. This allows you to use `validate` to ensure that the login flow is
+fully complete by, for example, repeatedly checking that the user's session
+token has been set. This must be done with a
+[retry-able command](/guides/core-concepts/retry-ability), or your assertion may
+fail the test simply because the page is running a little slow. Once your
+assertion passes, the session is cached.
+
+When restoring a saved session, if the `validate` function return `false`,
+throws an exception, returns a Promise that resolves to `false` or rejects, or
+contains any failing Cypress command, the session will be considered invalid,
+and `setup` will be re-run. So your `validate` function serves a dual purpose,
+it ensures that the initial setup has worked correctly **and** it checks that
+the session remains valid before re-use.
 
-The page is not cleared after the `validate` function is executed. If you use
-`cy.visit()` in your validation, your test will continue on the visited page
-once `cy.session()` succeeds.
+Note that the page is not cleared after the `validate` function is executed. If
+you use `cy.visit()` in your validation, your test will continue on the visited
+page once `cy.session()` succeeds.
```

Comment:
This is no longer true. When`testIsolation='on'`, we always clear before the command finishes. If `testIsolation='off'` the page will remain in tact.
```suggestion
```